src/boot/efi/util.c

```diff
@@ -390,10 +390,15 @@ void clear_screen(size_t attr) {
         ST->ConOut->ClearScreen(ST->ConOut);
 }
 
-void sort_pointer_array(
-                void **array,
+static void *array_element(const void *array, size_t stride, size_t i) {
+        return (uint8_t *) array + stride * i;
+}
```

Comment:
an overflow check for the multiplication would be great.

use MUL_SAFE() macro for example